### PR TITLE
Bug 2039277: fix topology list view

### PR DESCRIPTION
**Pull request by abhinandan13jan**
# Addresses
https://issues.redhat.com/browse/OCPBUGSM-39224
https://issues.redhat.com/browse/ODC-6163
https://bugzilla.redhat.com/show_bug.cgi?id=2039277
https://bugzilla.redhat.com/show_bug.cgi?id=2042466

# Issues / Solutions
**Accessibility issue:** Solved Topology related issue 
`<li>` must be directly put in `<ul>` or `<ol>` by updating `DatalistContent` to `DataList`.
Many issues are outside topology and `id` issue is because of `DataList` component and not `Topology`

**Highlight issue** 
Added `hover` css to `focus`

# Screenshots
**List Highlight and Selection** 
ezgif com-gif-maker

--------------------------------------------------------
**Accessibility issues post change**
Screenshot 2022-01-19 at 8 17 43 PM
---------------------------------------------------------
**Accessibilty issues currently on server** 
Screenshot 2022-01-19 at 8 18 18 PM
-------------------------------------------------------------

# Tests
no change

# Browser conformance
Chrome & Firefox

### CODE REVIEW

**Review by jerolimov — COMMENTED** [review]
Hi @abhinandan13jan, thanks for this PR. It's def. better than before. Esp. that the a11y error are gone :+1: 

But when navigating with the keyboard up-down keys, I can see that PatternFly already added a `pf-m-selected` and the sidebar content changes, but the element was not highlighted correctly. Is it possible to fix this together?

In this video, I press up and down keys to select the item. See changing sidebar content:

Peek 2022-01-24 10-46

In this video, you can see that PF already sets `pf-m-selected`:

Peek 2022-01-24 10-48

**Comment by abhinandan13jan** [comment]
> Hi @abhinandan13jan, thanks for this PR. It's def. better than before. Esp. that the a11y error are gone 👍
> 
> But when navigating with the keyboard up-down keys, I can see that PatternFly already added a `pf-m-selected` and the sidebar content changes, but the element was not highlighted correctly. Is it possible to fix this together?
> 
> In this video, I press up and down keys to select the item. See changing sidebar content:
> 
> Peek 2022-01-24 10-46
> 
> In this video, you can see that PF already sets `pf-m-selected`:
> 
> Peek 2022-01-24 10-48

Fixed issue
fix-topo-3

File: frontend/packages/topology/src/components/list-view/TopologyListViewAppGroup.tsx [hunk 1]
```diff
@@ -52,7 +52,7 @@ const TopologyListViewAppGroup: React.FC<TopologyListViewAppGroupProps> = ({
   cells.push(
     <DataListCell
       key={id}
-      className="odc-topology-list-view__application-label-cell"
+      className="odc-topology-list-view__application-label-cell odc-topology-list-view__nofocus-label"
```
Comment (jerolimov) [inline frontend/packages/topology/src/components/list-view/TopologyListViewAppGroup.tsx:55]: Don't find any css definition for `odc-topology-list-view__nofocus-label` ?! Can we drop this again?

```suggestion
      className="odc-topology-list-view__application-label-cell"
```

File: frontend/packages/topology/src/components/list-view/TopologyListViewUnassignedGroup.tsx [hunk 2]
```diff
@@ -53,7 +53,7 @@ const TopologyListViewUnassignedGroup: React.FC<TopologyListViewUnassignedGroupP
   cells.push(
     <DataListCell
       key="label"
-      className="odc-topology-list-view__unassigned-label"
+      className="odc-topology-list-view__unassigned-label odc-topology-list-view__nofocus-label"
```
Comment (jerolimov) [inline frontend/packages/topology/src/components/list-view/TopologyListViewUnassignedGroup.tsx:56]: Don't find any css definition for `odc-topology-list-view__nofocus-label` ?! Can we drop this again?

```suggestion
      className="odc-topology-list-view__unassigned-label"
```

File: frontend/packages/topology/src/components/list-view/TopologyListViewAppGroup.tsx (hunk 1 above)
Comment (abhinandan13jan) [reply frontend/packages/topology/src/components/list-view/TopologyListViewAppGroup.tsx]: updated

File: frontend/packages/topology/src/components/list-view/TopologyListViewUnassignedGroup.tsx (hunk 2 above)
Comment (abhinandan13jan) [reply frontend/packages/topology/src/components/list-view/TopologyListViewUnassignedGroup.tsx]: updated

**Comment by abhinandan13jan** [comment]
@jerolimov @rottencandy Please review
ezgif com-gif-maker

**Review by rottencandy — COMMENTED** [review]
Thanks @abhinandan13jan ,

Selecting deployment through keyboard navigation properly highlights the selected list item,
but selecting an application does not seem to highlight it. Anything we can do about it?

https://user-images.githubusercontent.com/20013884/151554165-193cef0f-6d2e-404f-9086-68b92227d5ae.mp4

**Comment by jerolimov** [comment]
It works much better than before.

* The lines between the items are restored
* The selected indicator on the left side of an item is back
* Mouse hover the items doesn't let the items below jump up and down anymore
* If fixes some accessibility issues

I see a small difference to 4.10 master that the padding around the Application group has changed.

diff

We also have another issue that the application group is not selectable anymore.

I would suggest that we merge this today and create new issues for things we had found / left over. Wdyt @rottencandy @abhinandan13jan ? It's definitely a step in the right direction, or do you have any concerns to get this in?

Maybe Jeff can also take a look?

/cc @jeff-phillips-18

**Review by rottencandy — APPROVED** [review]
Yes, I think we can get these changes in and create tickets for the other issues found.

/lgtm

**Review by jeff-phillips-18 — APPROVED** [review]

**Review by jerolimov — APPROVED** [review]

**Pull request merged**